corehq/apps/data_interfaces/tasks.py

```diff
@@ -111,3 +113,18 @@ def run_case_update_rules_for_domain(domain, now=None):
         for rule in rules:
             rule.last_run = now
             rule.save()
+
+
+@task(queue='background_queue', acks_late=True, ignore_result=True)
+def run_case_update_rules_on_save(domain, case_type, case_id):
+    rules = AutomaticUpdateRule.by_domain(domain).filter(case_type=case_type)
+    case = CaseAccessors(domain).get_case(case_id)
+    now = datetime.utcnow()
+    key = 'case-update-on-save-case-{case}'.format(case=case_id)
+    with CriticalSection([key]):
```

Comment:
i think you're seeing that because you must have run the function synchronously right? if run asynchronously it should just make the second task block until it's allowed to process while the first finishes, which is what we want